src/ansi-c/expr2c.cpp

```diff
@@ -3535,17 +3533,18 @@ std::string expr2ct::convert_with_precedence(
     return convert_byte_extract(to_byte_extract_expr(src), precedence = 16);
 
   else if(has_prefix(src.id_string(), "byte_update"))
-    return convert_byte_update(src, precedence=16);
+    return convert_byte_update(to_byte_update_expr(src), precedence = 16);
 
   else if(src.id()==ID_address_of)
   {
-    if(src.operands().size()!=1)
-      return convert_norep(src, precedence);
-    else if(src.op0().id()==ID_label)
-      return "&&"+src.op0().get_string(ID_identifier);
-    else if(src.op0().id()==ID_index &&
-            to_index_expr(src.op0()).index().is_zero())
-      return convert(to_index_expr(src.op0()).array());
+    const auto &object = to_address_of_expr(src).object();
+
+    if(object.id() == ID_label)
+      return "&&" + object.get_string(ID_identifier);
+    else if(object.id() == ID_index && to_index_expr(object).index().is_zero())
+    {
+      return convert(to_index_expr(object).array());
+    }
```

Comment:
Braces don't seem to be necessary here (I'm only objecting to their use, because the other cases of this `if` don't use braces either.